xbmc/filesystem/FileCache.cpp

```diff
@@ -212,7 +212,7 @@ void CFileCache::Process()
 
   // create our read buffer
   std::unique_ptr<char[]> buffer(new char[m_chunkSize]);
-  if (buffer.get() == NULL)
+  if (!buffer)
```

Comment:
This check seems to be redundant. If `new[]` fails, it should throw `std::bad_alloc` and we should never get to this line. If it succeeds, our pointer will be non-null.